pinot-core/src/main/java/org/apache/pinot/core/operator/docidsets/AndDocIdSet.java

```diff
@@ -52,9 +54,12 @@
  */
 public final class AndDocIdSet implements FilterBlockDocIdSet {
   private final List<FilterBlockDocIdSet> _docIdSets;
+  private final boolean _cardinalityBasedRankingForScan;
 
-  public AndDocIdSet(List<FilterBlockDocIdSet> docIdSets) {
+  public AndDocIdSet(List<FilterBlockDocIdSet> docIdSets, Map<String, String> queryOptions) {
     _docIdSets = docIdSets;
+    _cardinalityBasedRankingForScan = queryOptions != null
```

Comment:
(nit) IIRC, `queryOptions` should never be `null`. If it can be null, we should annotate it accordingly

Reply:
suggest check using MapUtils.isEmpty

Reply:
Moved QueryOptionsUtils. Yes it's not null.